### PR TITLE
[hotfix][engine][dag] Loss of parallelism when recreating actions.

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request
Loss of parallelism when recreating actions.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java [hunk 1]
```diff
@@ -208,6 +208,7 @@ public static Action recreateAction(Action action, Long id) {
         } else {
             throw new UnknownActionException(action);
         }
+        newAction.setParallelism(action.getParallelism());
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java:212]: Need to change line 174?

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/execution/ExecutionPlanGenerator.java:212]: +1

**Comment by ashulin** [comment]
@ic4y PTAL. I removed `Awaitility. Await` from the test class.

**Comment by ic4y** [comment]
> @ic4y PTAL. I removed `Awaitility. Await` from the test class.

why remove Awaitility. Await

**Comment by ashulin** [comment]
> > @ic4y PTAL. I removed `Awaitility. Await` from the test class.
> 
> why remove Awaitility. Await

Because Awaitility.await() will cause a timeout exception, making the unit test fail. 

The estimated timeout time is inaccurate, so I use `future#get()` and `CompletableFuture.allOf().join()` to wait the result of the future. If there is no future that is blocked all the time, there is no problem in removing it.

**Comment by ic4y** [comment]
> > > @ic4y PTAL. I removed `Awaitility. Await` from the test class.
> > 
> > 
> > why remove Awaitility. Await
> 
> Because Awaitility.await() will cause a timeout exception, making the unit test fail.
> 
> The estimated timeout time is inaccurate, so I use `future#get()` and `CompletableFuture.allOf().join()` to wait the result of the future. If there is no future that is blocked all the time, there is no problem in removing it.

1. If there is no timeout mechanism. Then when there is a bug in the program, the feature may be blocked all the time.
2. There are some tests when the test execution time is less than a fixed value (a value that is not very precise, but also has practical significance), which also requires a timeout mechanism.

**Pull request merged**